app/src/main/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -1898,232 +1905,14 @@ public void onLoadFinished(Loader<Cursor> loader, final Cursor data) {
-                                        } catch (ParseException e) {
-                                            e.printStackTrace();
-                                            // we need to persist data again
-
-                                            cloudStorageOnedrive.login();
-                                            cloudEntryOnedrive = new CloudEntry(OpenMode.ONEDRIVE, cloudStorageOnedrive.saveAsString());
-                                            cloudHandler.updateEntry(OpenMode.ONEDRIVE, cloudEntryOnedrive);
-                                        }
-                                    } else {
-                                        cloudStorageOnedrive.login();
-                                        cloudEntryOnedrive = new CloudEntry(OpenMode.ONEDRIVE, cloudStorageOnedrive.saveAsString());
-                                        cloudHandler.addEntry(cloudEntryOnedrive);
-                                    }
-
-                                    dataUtils.addAccount(cloudStorageOnedrive);
-                                    hasUpdatedDrawer = true;
-                                } catch (CloudPluginException e) {
-                                    e.printStackTrace();
-                                    AppConfig.toast(MainActivity.this, getResources().getString(R.string.cloud_error_plugin));
-                                    deleteConnection(OpenMode.ONEDRIVE);
-                                    return false;
-                                } catch (AuthenticationException e) {
-                                    e.printStackTrace();
-                                    AppConfig.toast(MainActivity.this, getResources().getString(R.string.cloud_fail_authenticate));
-                                    deleteConnection(OpenMode.ONEDRIVE);
-                                    return false;
-                                } catch (Exception e) {
-                                    // any other exception due to network conditions or other error
-                                    e.printStackTrace();
-                                    AppConfig.toast(MainActivity.this, getResources().getString(R.string.failed_cloud_new_connection));
-                                    deleteConnection(OpenMode.ONEDRIVE);
-                                    return false;
-                                }
-                                break;
-                            default:
-                                Toast.makeText(MainActivity.this, getResources().getString(R.string.cloud_error_failed_restart),
-                                        Toast.LENGTH_LONG).show();
-                                return false;
-                        }
-                    } while (data.moveToNext());
-                }
-                return hasUpdatedDrawer;
-            }
+        if (mCloudCursorData != null && mCloudCursorData == data) return;
+        mCloudCursorData = data;
```

Comment:
> [...]
> The Loader will release the data once it knows the application
> is no longer using it.  For example, if the data is
> a {@link android.database.Cursor} from a {@link android.content.CursorLoader},
> you should not call close() on it yourself.  If the Cursor is being placed in a
> {@link android.widget.CursorAdapter}, you should use the
> {@link android.widget.CursorAdapter#swapCursor(android.database.Cursor)}
> method so that the old Cursor is not closed.
> [...]

-- [onLoadFinished() documentation](https://developer.android.com/reference/android/app/LoaderManager.LoaderCallbacks.html#onLoadFinished(android.content.Loader%3CD%3E,%20D))

Reply:
We don't need the old cursor, this is just a hack to know if the request is a duplicate one. I couldn't find a proper solution to this for the time being.

Reply:
It doesn't say anywhere that this is a hack, and why it was implemented, what other solutions would be and why they don't work.

Reply:
onLoadFinished gets fired up again and again, without any reason. Simply try adding any cloud connection, pause Amaze and resume, onLoadFinished will get called for no reason. 
I figured this issue is a release blocker, also I am quite low on time myself.
So I wrote this to delegate these duplicate callbacks and this gets the job done. 
If you can find a real solution to this, I'll gladly accept your PR,

Reply:
I had made a [patch](https://gist.github.com/EmmanuelMess/8aabccfc7e3c213b33f03873d58d8cef) a while ago for an older version of master, what did you think of it. Because of the nature of the patch I couldn't know if it was just me not being able to replicate, or the issue being fixed.

Reply:
> If you can find a real solution to this, I'll gladly accept your PR,

I have 6 PRs stuck, please start reviewing if you want my help with this. I will start working on this ASAP if you review them.